### PR TITLE
Removes AOE hit detection from fireball

### PR DESCRIPTION
**Pull request by GoblinBackwards**
## About The Pull Request
Removes the AOE detection from fireball projectile which meant it would hit anyone it found on a tile next to it while flying

This has nothing to do with the fireballs explosion size. It's about how the fireball projectile itself calls bump() on any living player next to it.

hate fireball i ded 

## Why It's Good For The Game
Maybe giving a spell that can instantly crit people a 3 tile wide hitbox isn't a great idea, especially when 90% of map design is 3 tile wide hallways where it becomes impossible to dodge when fired down the middle. The massive ass hitbox also means you will hit people directly next to you and blow yourself up if you use it next to another player, even if you weren't aiming at them. Or you can shoot it past a corner and direct hit someone where if you actually aimed at them it would have just hit the wall.

## Changelog

:cl:
balance: Fireball no longer has a 3 tile wide hitbox.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
> Maybe giving a spell that can instantly crit people a 3 tile wide hitbox isn't a great idea

what? its still a 1 tile hitbox. it will just explode larger if it does hit something

**Comment by BebeYoshi** [comment]
I have heard that yogstation fix to explosion damage might affect our codebase too, and it does mention fireball as well there, this would possibly be a better change/fix? 
https://github.com/yogstation13/Yogstation/pull/13983

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -447,7 +447,7 @@
 	qdel(chain)
 	. = ..()
 
-/obj/projectile/magic/aoe/fireball
+/obj/projectile/magic/fireball
```
Comment (BebeYoshi) [inline code/modules/projectiles/projectile/magic.dm:450]: Does this even change anything? The explosion values are the same, so it should still be an AOE?

Comment (GoblinBackwards) [reply code/modules/projectiles/projectile/magic.dm:450]: This change has nothing to do with the explosion, it's about the projectile itself hitting players to the sides of it.

**Comment by Livrah** [comment]
Ppl can be mad about "nerf", I'm happy not to blow myself up by accident.

**Comment by zxaber** [comment]
Nerf to hallway Kamehameha, buff to fireball skillshots.

**Comment by ATH1909** [comment]
So THAT'S why their hit detection was so jank.

**Comment by private-tristan** [comment]
does this mean sleeping carp can now deflect it 👀

**Comment by Shadowflame909** [comment]
Isn't the chance of the spell hitting a target you didnt mean it to hit the reason it was robeless, 1 point, and fast to fire?

It's literally the "cast this at a bad time and you lose" classic spell of wizarddom!

I thought it was intentional.

**Comment by GoblinBackwards** [comment]
> does this mean sleeping carp can now deflect it 👀

yeah

**Comment by ATH1909** [comment]
> > does this mean sleeping carp can now deflect it 👀
> 
> yeah

metal as hell

**Comment by Shadowflame909** [comment]
> it costs 2 points, and if you're a 1 vs. station antag like a wizard, hitting a target you didn't intend to hit is probably still to your benefit. also, the _way_ it hit targets you didn't intend to hit was super jank and unintuitive. like, I'm pretty sure someone standing one tile north of you would intercept fireballs cast directly to your east or something like that.

It doesnt cost 1 point anymore? Yeah I take back my downvote. What a janky noob-bait spell!

**Comment by ATH1909** [comment]
> It doesnt cost 1 point anymore? Yeah I take back my downvote. What a janky noob-bait spell!

mate, fireball is one of the three best spells in the game (time stop, jaunt, fireball), idk what you're on about

**Comment by Shadowflame909** [comment]
its literally the only spell a wizard has that they can accidentally crit themselves with just by firing it (Which is why I thought it was 1 point + robeless) but i digress

**Comment by JohnFulpWillard** [comment]
AOE is a double-edged blade right now, with AOE its easier to hit a person but it can backfire pretty bad by firing point blank, without it you need to aim better to hit your target or it will backfire by having to wait the cooldown.

At least without AOE it doesn't allow you to bypass anti-magic by firing at the wall next to the holy person, so I think this is the better version of it.

**Pull request merged**